### PR TITLE
RHBRMS-2564 : Org.unit/repo/project navigation missing in standalone authoring perspective

### PR DESCRIPTION
**Pull request by Rikkola**
https://issues.jboss.org/browse/GUVNOR-2331

NO_CONTEXT_NAVIGATION is no longer used and instead the code checks if the OU/REPO/PROJECT selector is set to visible in the URL. By default we show it.

### CODE REVIEW

**Comment by Rikkola** [comment]
@manstis @karreiro if either of you could please review.

File: kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/ActiveContextOptions.java [hunk 1]
```diff
@@ -77,12 +79,14 @@ public void init( final PlaceRequest placeRequest,
...
-        }
 
         if ( options.isEmpty() ) {
-            load( completeCommand );
+            load(new Command() {
+                @Override
+                public void execute() {
+                    completeCommand.execute();
+                }
+            });
```
Comment (karreiro) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/ActiveContextOptions.java]: @Rikkola Why do we need a new Command here?

File: kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/BaseViewPresenter.java [hunk 2]
```diff
@@ -161,8 +161,10 @@ public void update() {
 
         baseView.setNavType( getNavType() );
 
-        if ( activeOptions.isHeaderNavigationHidden() ) {
+        if (activeOptions.isHeaderNavigationHidden()) {
             baseView.hideHeaderNavigator();
+        }else{
+            baseView.showHeaderNavigator();
```
Comment (karreiro) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/BaseViewPresenter.java]: Code formatting :-)

File: kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/navigator/Explorer.java [hunk 3]
```diff
@@ -166,42 +213,30 @@ public void onClick( ClickEvent event ) {
...
+        this.repos.setText( ProjectExplorerConstants.INSTANCE.nullEntry() );
+        this.repos.add( new AnchorListItem( ProjectExplorerConstants.INSTANCE.nullEntry() ) );
+        showEmptyProject();
+    }
 
-            isAlreadyInitialized = true;
-        }
+    private void showEmptyProject() {
+        this.projects.setText( ProjectExplorerConstants.INSTANCE.nullEntry() );
+        this.projects.add( new AnchorListItem( ProjectExplorerConstants.INSTANCE.nullEntry() ) );
```
Comment (karreiro) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/navigator/Explorer.java]: `showEmptyOU`, `showEmptyRepositories`, `showEmptyProject` :clap:

File: kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/ActiveContextOptions.java [hunk 4]
```diff
@@ -77,12 +79,14 @@ public void init( final PlaceRequest placeRequest,
             options.addAll( optionsFromModeParameter );
         }
 
-        if ( isContextNavigationOff( placeRequest ) ) {
-            options.add( Option.NO_CONTEXT_NAVIGATION );
-        }
 
         if ( options.isEmpty() ) {
-            load( completeCommand );
+            load(new Command() {
```
Comment (karreiro) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/ActiveContextOptions.java]: Code formatting :-)

File: kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/ActiveContextOptions.java (hunk 1 above)
Comment (Rikkola) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/ActiveContextOptions.java]: Good point. This is a left over from the previous fix I tried for this ticket and  forgot to change it back.

File: kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/navigator/Explorer.java (hunk 3 above)
Comment (Rikkola) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/navigator/Explorer.java]: Wanted to clean up more, but this fix will be cherry-picked to the previous branches. So decided to keep the changes small.

**Comment by Rikkola** [comment]
@karreiro Ok, back to you.

**Comment by karreiro** [comment]
+1

**Pull request merged**